### DISCUSSION TITLE
Conditional Update value with Case When Then

### DISCUSSION THREAD

**Discussion by sebarys** [slick/slick#2969] [Questions] [answered]
Hello,

I was wondering if it is possible to conditionally determine value for update statement. Basically I would construct given query
```
UPDATE dbo.TestStudents  
SET     LASTNAME =  CASE  
                        WHEN LASTNAME = 'AAA' THEN 'BBB' 
                        WHEN LASTNAME = 'CCC' THEN 'DDD' 
                        WHEN LASTNAME = 'EEE' THEN 'FFF' 
                        ELSE LASTNAME
                    END 
WHERE   LASTNAME IN ('AAA', 'CCC', 'EEE')
```
using Slick syntax.

based on `update` def:
```
    /** An Action that updates the data selected by this query. */
    def update(value: T): ProfileAction[Int, NoStream, Effect.Write] = {
```
I guess it isn't supported, but maybe someone know how it can be achieved?

Thanks in advance for any suggestions!

**Comment by nafg** [comment, marked as answer]
Currently the best way to achieve this within Slick is Mutators.

Apparently they are not currently documented on the website, but you can read this to learn about them:

 https://deepwiki.com/search/mutators_675257fb-8ff9-458c-89f2-02cfdbd0fcee?mode=fast
